conda-store-server/Dockerfile

```diff
@@ -1,7 +1,10 @@
-FROM --platform=linux/amd64 condaforge/mambaforge:23.3.1-1
+FROM --platform=linux/amd64 condaforge/mambaforge:23.3.1-1 as base
```

Comment:
Do we need to specify the `PLATFORM` here? this is now a deprecated flag and in any case we can use the build action to make platforms explicit https://github.com/docker/setup-buildx-action/tree/v1/

Reply:
No, we don't. I was a considering removing it, but don't like bundling too many changes into a PR, so wasn't sure if that change belonged here. I will remove it though.